doc/cookbook/multi_node_cluster.md

```diff
@@ -0,0 +1,229 @@
+  --name "machine-3" \
+  --initial-advertise-peer-urls http://$HOST3:2380 \
+  --listen-peer-urls http://$HOST3:2380 \
+  --listen-client-urls http://$HOST3:2379 \
+  --advertise-client-urls http://$HOST3:2379 \
+  --initial-cluster $CLUSTER
+```
+
+Now you can list cluster members
+```bash
+egctl --server $HOST1:2381 member list | grep " name"
+```
+should print 
+```bash
+    name: machine-1
+    name: machine-2
+    name: machine-3
+```
+
+###  Add new member
+
+Let's add one more node with a *secondary* cluster role this time.
+
+```bash
+# on machine 4
+easegress-server \
+  --cluster-name "multi-node-cluster" \
+  --cluster-role "secondary" \
+  --name "machine-4" \
+  --primary-listen-peer-urls http://$HOST1:2380 \
+  --state-flag "existing"
+```
+Here `primary-listen-peer-urls` tells, where to find a *primary* cluster member and `state-flag` with value "existing" means that this cluster was already created.
+
+We can now see also the 4th instance:
+```bash
+egctl --server $HOST1:2381 member list | grep " name"
+# prints
+    name: machine-1
+    name: machine-2
+    name: machine-3
+    name: machine-4
+```
+Congratulations, you now have your Easegress instances running! You can now start applying resources to Easegress, like [pipeline](./pipeline.md) or [workflow](./workflow.md) for example.
+
+You can also keep reading this tutorial to know more about YAML configuration of Easegress cluster instances or configuration tips.
+
+## YAML Configuration (optional)
+
+The examples above use the *easegress-server's* command line flags, but often it is more convenient to define server parameters in a yaml configuration file. For example, store following yaml to each host machine and change the host addresses accordingly.
+
+```yaml
+# create one yaml file for each host
+name: machine-1 # machine-2, machine-3
+cluster-name: cluster-test
+cluster-role: primary
+api-addr: localhost:2381
+data-dir: ./data
+wal-dir: ""
+cpu-profile-file:
+memory-profile-file:
+log-dir: ./log
+debug: false
+cluster:
+  listen-peer-urls: # change CURRENT-HOST to current host
+   - http://<CURRENT-HOST>:2380
+  listen-client-urls:
+   - http://<CURRENT-HOST>:2379
+  advertise-client-urls:
+   - http://<CURRENT-HOST>:2379
+  initial-advertise-peer-urls:
+   - http://<CURRENT-HOST>:2380
+  initial-cluster: # initial-cluster is same for every host
+   - machine-1: http://<HOST-1>:2380
+   - machine-2: http://<HOST-2>:2380
+   - machine-3: http://<HOST-3>:2380
+```
+Then apply these values on each machine, using `config-file` command line argument:
+`easegress-server --config-file config.yaml`.
+
+The configuration file for adding new secondary node looks like following:
+
+```yaml
+name: machine-4
+cluster-name: cluster-test
+cluster-role: secondary
+data-dir: ./data
+wal-dir: ""
+cpu-profile-file:
+memory-profile-file:
+log-dir: ./log
+debug: false
+cluster:
+  primary-listen-peer-urls: http://$HOST1:2380
+```
+
+## Configuration tips (optional)
```

Comment:
```suggestion
## Configuration tips
```

Reply:
Thanks for the comments! I'll keep `optional` here as this is not required for the tutorial.